HerPublicWebsite.BusinessLogic/Services/SessionRecorder/SessionRecorderService.cs

```diff
@@ -18,12 +20,21 @@ public SessionRecorderService(IDataAccessProvider dataAccessProvider, IDateTimeP
         this.dateTimeProvider = dateTimeProvider;
     }
 
-    public async Task RecordNewSessionStarted()
+    public async Task<Session> RecordNewSessionStarted()
     {
         var session = new Session
         {
             Timestamp = dateTimeProvider.Now()
         };
-        await dataAccessProvider.PersistSession(session);
+        return await dataAccessProvider.PersistSession(session);
+    }
+
+    public async Task SetIsJourneyCompleteToTrue(Questionnaire questionnaire)
+    {
+        var sessionId = questionnaire.SessionId;
+        if (sessionId != null)
```

Comment:
Feels like we should know this isn't the case by this point? Should we be throwing something if not?